### PR TITLE
Allow setting curator runhour/runminute timezone - default is UTC

### PR DESCRIPTION
**Pull request by richm**
https://github.com/openshift/origin-aggregated-logging/issues/182
Clarify in the documentation that the curator date/time settings
are with respect to the timezone of the curator pod, and change
the test to use that timezone.

### CODE REVIEW

**Comment by richm** [comment]
@xiazhao2015 @sosiouxme @ewolinetz PTAL

**Comment by richm** [comment]
[test]

**Comment by ewolinetz** [comment]
LGTM

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
@ewolinetz @sosiouxme @smarterclayton  what scc should I use for the curator user?  fluentd does this in order to mount `/var/log` and `/var/lib`:
```
os::cmd::expect_success "oadm policy add-scc-to-user privileged system:serviceaccount:logging:aggregated-logging-fluentd"
```
But `privileged` seems too much - all curator needs to do is mount `/etc/localtime`.

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]

**Comment by portante** [comment]
So ES stores documents in UTC.  If the @timestamp field has a timezone in it, the timestamp value will be converted to UTC and then stored.

The convention about index names is that they reflect the range of timestamps stored in them.  So if they are not created using UTC values, then one could be loosing data unexpectedly.

Why not ensure all index names are created using UTC values and make sure the curator times are specified only in UTC?

**Comment by richm** [comment]
@portante That's a separate issue.  This issue is about when the curator "cron" job runs.  We should open a new issue - ensure fluentd writes `@timestamp` and timestamps in index names using UTC.

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
@ewolinetz @sosiouxme @jcantrill  @xiazhao2015 PTAL

**Comment by richm** [comment]
@ewolinetz @sosiouxme @jcantrill @xiazhao2015 final comments?

File: hack/testing/test-curator.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-if [[ $VERBOSE ]]; then
+if [ -n "${VERBOSE:-}" ] ; then
```
Comment (ewolinetz) [inline hack/testing/test-curator.sh]: [ "$VERBOSE" = true ]  to match check-logs.sh

**Comment by ewolinetz** [comment]
nit otherwise LGTM

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
[test]
sigh - etcd issues again . . .

**Comment by richm** [comment]
[test]
grr - etcd issues again . . .

**Comment by richm** [comment]
@ewolinetz @sosiouxme @jcantrill @xiazhao2015 final comments?

File: curator/run_cron.py [hunk 2]
```diff
@@ -31,6 +42,15 @@
 with open(filename, 'r') as stream:
     decoded = yaml.load(stream) or {}
 
+tzstr = decoded.get('.defaults', {}).get('timezone', os.getenv('CURATOR_RUN_TIMEZONE', 'UTC'))
+tz = None
+if tzstr:
+    try:
+        tz = timezone(tzstr)
+    except:
+        logger.error('The timezone must be specified in the tzselect(8) or timedatectl(1) "Region/Locality" format e.g. "America/New_York" or "UTC".')
```
Comment (sosiouxme) [inline curator/run_cron.py]: Could the error indicate what tzstr was and what the exception was?

File: deployer/README.md [hunk 3]
```diff
@@ -547,18 +552,24 @@ configuration in the deployed configmap:
 
     $ oc edit configmap/logging-curator
 
-Since it can be tricky to get YAML indentation correct in this context, you may
-prefer to create a yaml file with your configuration settings using your favorite editor.
-Next create a secret from your created yaml file:
+If this does not redeploy automatically, redeploy manually:
```
Comment (sosiouxme) [inline deployer/README.md]: It doesn't redeploy automatically

File: deployer/README.md [hunk 4]
```diff
@@ -547,18 +552,24 @@ configuration in the deployed configmap:
...
 
-Then overwrite the configuration with your created secret as a volume in your Curator DC:
+For scripted deployments, do this:
 
-    $ oc set volume dc/logging-curator --overwrite --name=config \
-             --type=secret --secret-name=index-management
-    $ oc deploy --latest logging-curator
+    $ create /path/to/mycuratorconfig.yaml
+    $ oc delete configmap logging-curator ; sleep 1
+    $ oc create configmap logging-curator --from-file=config.yaml=/path/to/mycuratorconfig.yaml ; sleep 1
```
Comment (sosiouxme) [inline deployer/README.md]: I prefer to replace the configmap seamlessly e.g.:

```
$ oc create configmap logging-curator -o yaml \
      --from-file=config.yaml=/path/to/mycuratorconfig.yaml | \
      oc replace -f -
```

**Comment by sosiouxme** [comment]
nits, LGTM otherwise

**Comment by richm** [comment]
[merge]

**Pull request merged**